agent/taskresource/firelens/firelens_unix.go

```diff
@@ -406,6 +489,54 @@ func (firelens *FirelensResource) generateConfigFile() error {
 	return nil
 }
 
+func (firelens *FirelensResource) downloadConfigFromS3() error {
```

Comment:
Just for my own understanding: This call comes in the FireLens task resource, which means it blocks the firelens container and any dependent containers from launching, but otherwise its asynchronous, so the agent can launch other containers/etc while the download is happening. Is that correct?

Reply:
yes, it only blocks the firelens container and containers that use awsfirelens driver. other containers can start running before/during the download.